pghoard/restore.py

```diff
@@ -138,19 +163,36 @@ def get_basebackup(self, arg):
         """Download a basebackup from an object store"""
         try:
             self.storage = self._get_object_storage(arg.config, arg.site, arg.target_dir)
-            self._get_basebackup(arg.target_dir, arg.basebackup, arg.site, arg.primary_conninfo, overwrite=arg.overwrite)
+            self._get_basebackup(arg.target_dir, arg.basebackup, arg.site, arg.primary_conninfo,
+                                 recovery_target_time=arg.recovery_target_time,
+                                 recovery_target_xid=arg.recovery_target_xid,
+                                 overwrite=arg.overwrite)
         except Exception as ex:
             raise RestoreError("{}: {}".format(ex.__class__.__name__, ex))
 
-    def _get_basebackup(self, pgdata, basebackup, site, primary_conninfo, overwrite=False):
+    def _find_nearest_basebackup(self, recovery_target_time=None):
+        basebackups = self.storage.list_basebackups()
+        if recovery_target_time:
+            applicable_basebackups = [basebackup["name"] for basebackup in basebackups
```

Comment:
This considers both `name` AND `last_modified` to decide which backup to use - I think we should just use one field to select the backup to use.

Also it's not quite clear what `last_modified` means as it comes from the object store, it can be the time the upload finished, the time someone updated metadata or the time we started the upload so it's not clear to me that we can always rely on it.  OTOH name only includes date so we can't really use it alone either.  IMO we should modify backups' names to include full ISO8601 (basic) timestamps and only consider them when selecting the backup, that way we could be sure we're recovering to a specific point in time.